lib/std/target.zig

```diff
@@ -974,135 +974,11 @@ pub const Target = struct {
-                    .loongarch32 => .NONE,
-                    .loongarch64 => .NONE,
-                };
-            }
-
-            pub fn toCoffMachine(arch: Arch) std.coff.MachineType {
-                return switch (arch) {
-                    .avr => .Unknown,
-                    .msp430 => .Unknown,
-                    .arc => .Unknown,
-                    .arm => .ARM,
-                    .armeb => .Unknown,
-                    .dxil => .Unknown,
-                    .hexagon => .Unknown,
-                    .m68k => .Unknown,
-                    .le32 => .Unknown,
-                    .mips => .Unknown,
-                    .mipsel => .Unknown,
-                    .powerpc, .powerpcle => .POWERPC,
-                    .r600 => .Unknown,
-                    .riscv32 => .RISCV32,
-                    .sparc => .Unknown,
-                    .sparcel => .Unknown,
-                    .tce => .Unknown,
-                    .tcele => .Unknown,
-                    .thumb => .Thumb,
-                    .thumbeb => .Thumb,
-                    .x86 => .I386,
-                    .xcore => .Unknown,
-                    .xtensa => .Unknown,
-                    .nvptx => .Unknown,
-                    .amdil => .Unknown,
-                    .hsail => .Unknown,
-                    .spir => .Unknown,
-                    .kalimba => .Unknown,
-                    .shave => .Unknown,
-                    .lanai => .Unknown,
-                    .wasm32 => .Unknown,
-                    .renderscript32 => .Unknown,
-                    .aarch64_32 => .ARM64,
-                    .aarch64 => .ARM64,
-                    .aarch64_be => .ARM64,
-                    .mips64 => .Unknown,
-                    .mips64el => .Unknown,
-                    .powerpc64 => .Unknown,
-                    .powerpc64le => .Unknown,
-                    .riscv64 => .RISCV64,
-                    .x86_64 => .X64,
-                    .nvptx64 => .Unknown,
-                    .le64 => .Unknown,
-                    .amdil64 => .Unknown,
-                    .hsail64 => .Unknown,
-                    .spir64 => .Unknown,
-                    .wasm64 => .Unknown,
-                    .renderscript64 => .Unknown,
-                    .amdgcn => .Unknown,
-                    .bpfel => .Unknown,
-                    .bpfeb => .Unknown,
-                    .csky => .Unknown,
-                    .sparc64 => .Unknown,
-                    .s390x => .Unknown,
-                    .ve => .Unknown,
-                    .spu_2 => .Unknown,
-                    .spirv32 => .Unknown,
-                    .spirv64 => .Unknown,
-                    .loongarch32 => .Unknown,
-                    .loongarch64 => .Unknown,
-                };
-            }
+            // TODO: remove after 0.11.0 release
+            /// deprecated: use std.elf.EM.fromTargetCpuArch instead
+            pub const toElfMachine = "read the doc comment of this declaration";
```

Comment:
Could we not have a `@compileError` with the actual error message in here?

Reply:
That's what the previous commit did. Doesn't work because of this: https://github.com/ziglang/zig/pull/15885#issuecomment-1566242652
It's triggered immediately.